File: src/client/java/com/somefrills/features/farming/AutoWarpHome.java
package com.somefrills.features.farming;

import com.somefrills.config.FrillsConfig;
import com.somefrills.events.ServerJoinEvent;
import com.somefrills.events.TabListUpdateEvent;
import com.somefrills.features.core.AreaFeature;
import com.somefrills.misc.Area;
import com.somefrills.misc.Utils;
import meteordevelopment.orbit.EventHandler;

import java.util.List;

public class AutoWarpHome extends AreaFeature {
    private static PestStatus lastStatus = PestStatus.UNKNOWN; // -1 = unknown

    public AutoWarpHome() {
        super(FrillsConfig.instance.farming.autoWarpHomeEnabled);
    }

    private static PestStatus checkAliveState(List<String> tabListLines) {
        for (String line : tabListLines) {
            if (line.contains("Alive")) {
                return line.contains("0") ? PestStatus.CLEARED : PestStatus.PRESENT;
            }
        }
        return PestStatus.UNKNOWN;
    }

    @EventHandler
    private void onWorldTick(TabListUpdateEvent event) {
        PestStatus status = checkAliveState(event.lines);
        if (status == PestStatus.UNKNOWN) return;

        // First valid reading → just store it
        if (lastStatus == PestStatus.UNKNOWN) {
            lastStatus = status;
            return;
        }

        // Trigger only on transition >0 → 0
        if (status == PestStatus.CLEARED && lastStatus == PestStatus.PRESENT) {
            Utils.info("AutoWarpHome: >0 → 0 detected, running /home");
            Utils.runCommand("home");
        }

        lastStatus = status;
    }

    @EventHandler
    private void onJoin(ServerJoinEvent event) {
        lastStatus = PestStatus.UNKNOWN;
    }

    @Override
    protected boolean checkArea(Area area) {
        return area.equals(Area.GARDEN) && Utils.isOnGardenPlot();
    }

    public enum PestStatus {
        UNKNOWN, PRESENT, CLEARED
    }
}

File: src/main/kotlin/com/somefrills/features/farming/AutoWarpHome.kt
package com.somefrills.features.farming

import com.somefrills.config.FrillsMod

import com.somefrills.events.ServerJoinEvent
import com.somefrills.events.TabListUpdateEvent
import com.somefrills.features.core.AreaFeature
import com.somefrills.misc.Area
import com.somefrills.modules.FrillsFeature
import com.somefrills.utils.ChatUtils
import com.somefrills.utils.PlayerUtils
import com.somefrills.utils.SkyblockUtils
import com.somefrills.events.core.EventHandle

@FrillsFeature
object AutoWarpHome : AreaFeature(FrillsMod.config.farming.autoWarpHomeEnabled) {
    private var lastStatus: PestStatus? = PestStatus.UNKNOWN // -1 = unknown

    private fun checkAliveState(tabListLines: MutableList<String>): PestStatus {
        for (line in tabListLines) {
            if (line.contains("Alive")) {
                return if (line.contains("0")) PestStatus.CLEARED else PestStatus.PRESENT
            }
        }
        return PestStatus.UNKNOWN
    }

    @EventHandle
    private fun onWorldTick(event: TabListUpdateEvent) {
        val status: PestStatus = checkAliveState(event.lines)
        if (status == PestStatus.UNKNOWN) return

        // First valid reading → just store it
        if (lastStatus == PestStatus.UNKNOWN) {
            lastStatus = status
            return
        }

        // Trigger only on transition >0 → 0
        if (status == PestStatus.CLEARED && lastStatus == PestStatus.PRESENT) {
            ChatUtils.info("AutoWarpHome: >0 → 0 detected, running /home")
            PlayerUtils.runCommand("home")
        }

        lastStatus = status
    }

    @EventHandle
    private fun onJoin(event: ServerJoinEvent) {
        lastStatus = PestStatus.UNKNOWN
    }

    override fun checkArea(area: Area): Boolean {
        return area == Area.GARDEN && SkyblockUtils.isOnGardenPlot()
    }

    enum class PestStatus {
        UNKNOWN, PRESENT, CLEARED
    }
}